### ISSUE TITLE
Integration of FLUKA with DAGMC in CI

### ISSUE THREAD

**Issue by ahnaf-tahmid-chowdhury** [svalinn/DAGMC#955]
## Description

It has been noticed that DAGMC can use FLUKA as an optional dependency. However, our current CI setup does not include any methods to handle this integration.

## Details
- **Dependency:** FLUKA (optional for DAGMC)
- **Current CI Status:** No methods implemented for handling FLUKA
- **Documentation:** Build instructions for FLUKA available

## Actions Required
1. Implement methods in CI to handle FLUKA integration
2. Create test scripts for FLUKA

**Comment by gonuke** [comment]
Thanks @ahnaf-tahmid-chowdhury - this wasn't here originally because FLUKA was not always open source.  Now that it is, this will be a good additino.

**Comment by ahnaf-tahmid-chowdhury** [comment]
I found that we need to register to obtain access to the FLUKA binary. Additionally, it's possible to request access to the source code under specific conditions. I have a license for NukeHub, but I believe we'll require a separate license for DagMC for CI integration. Please let me know if you're interested in proceeding with this.

**Comment by gonuke** [comment]
FLUKA forked a few years ago and there is now a fully open version maintained by CERN, in addition to the managed-access version maintained by others.  The purpose of #677 is to change our FLUKA support to work with that version.

I am not sure your license to the managed-access version allows you to make it available to others via NukeHub - you should check.

**Comment by gonuke** [comment]
I think we want to leave the managed-access version behind and rely only on the open source version.

**Comment by ahnaf-tahmid-chowdhury** [comment]
Yes, I do have the same doubt. Thus, I haven't integrated the FLUKA with the NukeHub. I am planning to ask FLUKA team about this.

However, I tried to find an open version of FLUKA but unfortunately didn't find any. The only option I found is through registration. I would be grateful if you could share the link to the open version.

**Comment by gonuke** [comment]
I think I was wrong - I know there was a forking of the development tree, but I thought one became more open.  Perhaps that never happened or has been reversed?  I wonder if @makeclean van shed more light on this?
